includes/class-freemius.php

```diff
@@ -17790,6 +17385,15 @@ function opt_in(
 
                 $this->maybe_modify_api_curl_error_message( $result );
 
+                $this->store_connectivity_info(

```

Comment:
@fajardoleo it looks like the code assumes the error returned from the API is a connectivity issue. What if the error is related to something else?

Reply:
@vovafeldman From what I'm seeing, in `/wp-includes/class-wp-http.php`, these are the only possible errors:
- $response = new WP_Error( 'http_request_failed', __( 'A valid URL was not provided.' ) );
- $response = new WP_Error( 'http_request_not_executed', __( 'User has blocked requests through HTTP.' ) );

So the current logic supports both. In case there's a connection, the error returned from the API will be handled like before.

Reply:
I see 👍